### PR TITLE
[CLOUDTRUST-4637] Support creation/retrieval of SAML persistent name ID with user profiles

### PR DESCRIPTION
**Pull request by fperot74**

### CODE REVIEW

File: keycloak-rest-api-extensions-tests/pom.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@
         <dependency>
             <groupId>io.cloudtrust</groupId>
             <artifactId>kc-cloudtrust-test-tools</artifactId>
-            <version>${kc-cloudtrust.version}</version>
+            <version>18.0.0</version>
```
Comment (harture) [inline keycloak-rest-api-extensions-tests/pom.xml:33]: Why do we need to change this ?

File: keycloak-rest-api-extensions/pom.xml [hunk 2]
```diff
@@ -22,12 +22,11 @@
         <dependency>
             <groupId>io.cloudtrust</groupId>
             <artifactId>cloudtrust-common</artifactId>
-            <version>${cloudtrust.version}</version>
         </dependency>
         <dependency>
             <groupId>io.cloudtrust</groupId>
             <artifactId>kc-cloudtrust-common</artifactId>
-            <version>${kc-cloudtrust.version}</version>
+            <version>18.0.0</version>
```
Comment (harture) [inline keycloak-rest-api-extensions/pom.xml:29]: Same here

File: keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/admin/UsersResource.java [hunk 3]
```diff
@@ -77,20 +79,27 @@ public UsersResource(KeycloakSession session, AdminPermissionEvaluator auth, Adm
...
-            }
+            groups.forEach(user::joinGroup);
 
             // Add roles
-            for (RoleModel role : roles) {
-                user.grantRole(role);
+            roles.forEach(user::grantRole);
+
+            if (BooleanUtils.isTrue(generateNameID)) {
+                user.setSingleAttribute(CtUserResource.ATTRIB_NAME_ID, "G-" + UUID.randomUUID());
```
Comment (harture) [inline keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/admin/UsersResource.java:102]: Is it really an uppercase G by default ?

**Review by harture — APPROVED** [review]

File: keycloak-rest-api-extensions-tests/pom.xml (hunk 1 above)
Comment (fperot74) [reply keycloak-rest-api-extensions-tests/pom.xml:33]: Discussed by phone

**Pull request merged**